src/applications/mhv/landing-page/components/NavCard.jsx

```diff
@@ -11,12 +10,14 @@ const NavCard = ({ icon = null, title, links }) => {
         href={href}
         aria-label={ariaLabel}
         onClick={() => {
-          const destinationDomain = getDestinationDomain(href);
+          const header =
+            typeof text === 'object' && text !== null ? 'Inbox' : text;
```

Comment:
Would something like `text?.innerText ? innerText : String(text)` work? That should work for the more general case where the object is an HTMLElement that we can grab the rendered text from. There are probably refinements needed, but I think that optional chain syntax `?.` lets us check if the variable has the innerText prop, and use that if so. `String(text)` may be unnecessary.

I ran `$$("li.mhv-c-navlistitem").map(l => l?.textContent ? l.textContent : l)` in Firefox and Chrome to see what this spits out, and I see "Inbox" in the array of textContents. Will also do a quick check in React code, in case something is different within JSX

Reply:
`event.target` might be the right approach, rather than the jsx/React object, which doesn't play nice when unrendered.

It also may be we want to revisit how the links are generated/resolved, given there may be more red dots in the page's future. Past me was shortsighted.

Reply:
Is the dot notification used for more than just the email? I thought it was just for messages, how the "dot hack" was setup before was to just check if text = text and then if so then show the text....and if text didn't equal text then show 'Inbox'. So, it wasn't exactly "robust" before. When the dot is there it's an object so thats why I check for objects and not null. I guess I have never seen the dot notification anywhere but the messages.

Reply:
That makes sense, I understand now. I would feel better fixing it now while the context is fresh then to save it for later. I just put together something a bit more flexible in my most recent commit

Reply:
Per conversations, we're going to address this more holistically via a refactor ticket